src/plonk/mod.rs

```diff
@@ -338,8 +332,17 @@ impl<C: CurveAffine> RelaxedPlonkInstance<C> {
             .zip(&U2.instance)
             .map(|(a, b)| *a + *r * b)
             .collect::<Vec<C::ScalarExt>>();
-        let u = self.u + *r;
-        let y = self.y + *r * U2.y;
+
+        let challenges = self
+            .challenges
+            .iter()
+            .zip(
+                U2.challenges
+                    .iter()
+                    .chain(std::iter::once(&C::ScalarExt::ONE)),
+            )
+            .map(|(a, b)| *a + *r * b)
+            .collect::<Vec<C::ScalarExt>>();
```

Comment:
If the size of one of the challenges is smaller, the smallest one will be taken. This does not seem to be the expected behavior.

Maybe [zip_longest](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.zip_longest) or [zip_eq](https://docs.rs/itertools/latest/itertools/fn.zip_eq.html)?

Reply:
Suppose the challenges in PlonkInstance is [y1,y2]. Then the folded challenges in RelaxedInstance would be [y1,y2,u] (where u is the homogenous variable).  Since we don't put the default F::ONE in PlonkInstance, when we fold them, we just put it at the end. This is the same behavior as before.

Reply:
Well then we need to add pre-checking of len, since any implicit behavior can be violated during modification. 

That is, if it were an array, the restriction would be explicit and the compiler would check it for us, but now we have to take care of it at least this way.

Reply:
will use zip_eq here